python_scripts/load_arrow.py

```diff
@@ -379,7 +379,8 @@ def make_vars(self, pre_processed_column_types):
             # variables with the appropriate typing based on the vartype
             for varname, vartype in self.column_types.items():
                 if vartype == "string":
-                    sfi.Data.addVarStrL(varname)
+                    max_length = max(len(val.as_py()) for (val) in batch[varname])
```

Comment:
I guess this has the same potential edge case as with numeric ranges where the first batch might not be representative. But again, I think it's unlikely and as long as the failure more is an explicit error rather than silent corruption then I think it's OK.

Reply:
I'm not sure this is explicitly checked in tests, but I _think_ stata just recasts strings if they're not big enough - I'll look at adding a test in my pytest PR to follow :)